sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -497,6 +498,11 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
       StructType(schema.filterNot(_.name == parsedOptions.columnNameOfCorruptRecord))
 
     val linesWithoutHeader: RDD[String] = maybeFirstLine.map { firstLine =>
+      if (parsedOptions.enforceSchema == false) {
+        CSVDataSource.checkHeader(firstLine, new CsvParser(parsedOptions.asParserSettings),
```

Comment:
The function `checkHeader` is also called in `readFile`. Is it possible that it is called twice?